pkg/storage/cloud/external_storage.go

```diff
@@ -518,39 +556,74 @@ func checkHTTPContentRangeHeader(h string, pos int64) error {
 	return nil
 }
 
+func (r *resumingHTTPReader) sendRequest(
+	reqHeaders map[string]string,
+) (resp *http.Response, err error) {
+	prev := timeutil.Now()
+	attempt := 0
```

Comment:
you could declare (and inc) this as part of the `for` loop syntax if you felt like it.